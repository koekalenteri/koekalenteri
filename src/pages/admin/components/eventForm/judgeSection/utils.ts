import type { DeepPartial, EventClass, EventType, Judge, PublicJudge } from '../../../../../types'
import type { PartialEvent } from '../../EventForm'

import { isSameDay } from 'date-fns'

export type PartialPublicJudge = Partial<PublicJudge>
export type PartialPublicJudgeValue = PartialPublicJudge | PartialPublicJudge[]

export const filterJudges = (
  judges: Judge[],
  eventJudges: PublicJudge[],
  id: number | undefined,
  eventType?: EventType
) =>
  judges
    .filter((j) => !eventType?.official || j.eventTypes.includes(eventType.eventType))
    .filter((j) => j.id === id || !eventJudges.find((ej) => ej.id === j.id))

export const hasJudge = (c: DeepPartial<EventClass>, id?: number) =>
  Array.isArray(c.judge) ? c.judge.find((j) => j.id === id) : c.judge?.id === id

export const filterClassesByJudgeId = (classes?: EventClass[], id?: number) => classes?.filter((c) => hasJudge(c, id))

export const toArray = (j?: PartialPublicJudge): PartialPublicJudge[] => (j ? [j] : [])
export const makeArray = (j?: PartialPublicJudgeValue) => (Array.isArray(j) ? [...j] : toArray(j))
export const selectJudge = (j?: PartialPublicJudgeValue, judge?: PublicJudge): PartialPublicJudge[] => {
  const a = makeArray(j)
  if (judge) {
    const index = a.findIndex((cj) => cj.id === judge.id)
    if (index === -1) {
      a.push(judge)
    } else {
      a[index] = judge
    }
  }
  return a
}

export const removeJudge = (j?: PartialPublicJudgeValue, id?: number): PartialPublicJudge[] => {
  const a = makeArray(j)
  return a.filter((cj) => cj.id !== id)
}

export const updateJudge = (
  event: PartialEvent,
  id: number | undefined,
  judge: PublicJudge | undefined,
  values?: DeepPartial<EventClass>[]
) => {
  const isSelected = (c: DeepPartial<EventClass>) =>
    values?.find((v) => event && isSameDay(v.date ?? event.startDate, c.date ?? event.startDate) && v.class === c.class)
  return event.classes?.map((c) => ({
    ...c,
    judge: isSelected(c) ? selectJudge(c.judge, judge) : removeJudge(c.judge, id),
  }))
}
