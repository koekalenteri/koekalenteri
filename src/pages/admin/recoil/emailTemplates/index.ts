export * from './actions'
export * from './atoms'
export * from './atoms2'
export * from './effects'
export * from './selectors'
