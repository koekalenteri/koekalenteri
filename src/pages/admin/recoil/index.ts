export * from './emailTemplates'
export * from './events'
export * from './eventTypes'
export * from './officials'
export * from './organizers'
export * from './registrations'
export * from './user'
