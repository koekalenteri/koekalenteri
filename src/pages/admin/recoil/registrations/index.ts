export * from './atoms'
// 2nd file to avoid circular deps
export * from './atoms2'
export * from './selectors'
