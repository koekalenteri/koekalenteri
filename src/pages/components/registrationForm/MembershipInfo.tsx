import type { DeepPartial, Registration } from '../../../types'

import { useCallback } from 'react'
import { useTranslation } from 'react-i18next'
import Checkbox from '@mui/material/Checkbox'
import FormControlLabel from '@mui/material/FormControlLabel'
import FormGroup from '@mui/material/FormGroup'

import CollapsibleSection from '../CollapsibleSection'

import { useDogCacheKey } from './hooks/useDogCacheKey'

interface Props {
  readonly reg: DeepPartial<Registration>
  readonly disabled?: boolean
  readonly onChange?: (props: DeepPartial<Registration>) => void
  readonly orgId: string
}

const MembershipInfo = ({ reg, disabled, onChange, orgId }: Props) => {
  const { t } = useTranslation()
  const [ownerCache, setOwnerCache] = useDogCacheKey(reg.dog?.regNo, 'owner')
  const [handlerCache, setHandlerCache] = useDogCacheKey(reg.dog?.regNo, 'handler')

  const handleChange = useCallback(
    (props: DeepPartial<Pick<Registration, 'owner' | 'handler'>>) => {
      const changes = {}
      let cachedOwner, cachedHandler

      if (props.owner) {
        const membership =
          props.owner?.membership === undefined
            ? ownerCache?.membership
            : { ...ownerCache?.membership, [orgId]: props.owner.membership }
        cachedOwner = setOwnerCache({
          ...reg.owner,
          ownerHandles: reg.ownerHandles,
          ownerPays: reg.ownerPays,
          ...ownerCache,
          ...props.owner,
          membership,
        })
      }

      if (props.handler) {
        const membership =
          props.handler?.membership === undefined
            ? handlerCache?.membership
            : { ...handlerCache?.membership, [orgId]: props.handler.membership }
        cachedHandler = setHandlerCache({ ...reg.handler, ...handlerCache, ...props.handler, membership })
      }

      if (cachedOwner) {
        const { ownerHandles, ownerPays, ...owner } = cachedOwner
        Object.assign(changes, { owner: { ...owner, membership: owner.membership?.[orgId] } })
      }
      if (cachedHandler) {
        Object.assign(changes, { handler: { ...cachedHandler, membership: cachedHandler.membership?.[orgId] } })
      }

      if (Object.keys(changes).length) {
        onChange?.(changes)
      }
    },
    [handlerCache, onChange, orgId, ownerCache, setHandlerCache, setOwnerCache]
  )

  const open = !!reg.dog?.regNo

  return (
    <CollapsibleSection
      title={t('registration.membership')}
      open={open}
      error={!open}
      helperText={!open ? t('validation.registration.choose', { field: 'dog' }) : undefined}
    >
      <FormGroup>
        <FormControlLabel
          control={
            <Checkbox
              disabled={disabled}
              checked={reg.owner?.membership ?? false}
              onChange={(e) => handleChange({ owner: { membership: e.target.checked } })}
            />
          }
          label={t('registration.ownerIsMember')}
          name="ownerIsMember"
        />
        <FormControlLabel
          disabled={disabled || reg.ownerHandles}
          control={
            <Checkbox
              checked={reg.ownerHandles ? (reg.owner?.membership ?? false) : (reg.handler?.membership ?? false)}
              onChange={(e) => handleChange({ handler: { membership: e.target.checked } })}
            />
          }
          label={t('registration.handlerIsMember')}
          name="handlerIsMember"
        />
      </FormGroup>
    </CollapsibleSection>
  )
}

export default MembershipInfo
