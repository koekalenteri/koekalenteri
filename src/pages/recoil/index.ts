export * from './effects'
export * from './events'
export * from './judges'
export * from './registration'
export * from './user'
