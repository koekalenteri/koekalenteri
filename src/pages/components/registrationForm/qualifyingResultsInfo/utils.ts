import type { EventResultRequirement, EventResultRequirements, EventResultRequirementsByDate } from '../../../../rules'
import type { ManualTestResult, QualifyingResult } from '../../../../types'

import { nanoid } from 'nanoid'

import { unique } from '../../../../lib/utils'
import { objectContains } from '../validation'

const asArray = (v: EventResultRequirements | EventResultRequirement) => (Array.isArray(v) ? v : [v])

function findFirstMissing(requirements: EventResultRequirementsByDate | undefined, results: QualifyingResult[]) {
  if (!requirements || typeof requirements.rules === 'function') {
    return []
  }
  for (const rule of requirements.rules) {
    for (const opt of asArray(rule)) {
      const { count, ...rest } = opt
      if (results.filter((r) => objectContains(r, rest)).length < count) {
        return rest
      }
    }
  }
}

export function availableTypes(requirements?: EventResultRequirementsByDate, eventType?: string) {
  if (!requirements || typeof requirements.rules === 'function') {
    switch (eventType) {
      case 'NOME-B SM':
        return ['NOME-B']
      case 'NOME-A SM':
        return ['NOME-A', 'NOME-A KV']
      case 'NOWT SM':
        return ['NOWT']
    }
    return []
  }
  return unique(requirements.rules.flatMap((rule) => asArray(rule).map((opt) => opt.type)))
}

const availableSMResults = (type?: string, eventType?: string, hasKVA?: boolean) => {
  switch (eventType) {
    case 'NOME-B SM':
      return [hasKVA ? undefined : 'FI KVA-B', 'VOI1', 'VOI2', 'VOI3'].filter(Boolean)
    case 'NOWT SM':
      return [hasKVA ? undefined : 'FI KVA-WT', 'VOI1', 'VOI2', 'VOI3'].filter(Boolean)
    case 'NOME-A SM':
      if (type === 'NOME-A')
        return [hasKVA ? undefined : 'FI KVA-FT', 'A1 CERT', 'A1 RES-CERT', 'A1', 'A2', 'A3'].filter(Boolean)

      if (type === 'NOME-A KV') return ['EXC CACIT', 'EXC RES-CACIT', 'EXC', 'VG', 'G']
  }
  return []
}

export function availableResults(
  requirements?: EventResultRequirementsByDate,
  type?: string,
  eventType?: string,
  existingResults?: ManualTestResult[]
) {
  if (!requirements || typeof requirements.rules === 'function') {
    if (eventType?.includes(' SM')) {
      const hasKVA = existingResults?.some((r) => r.result.startsWith('FI KVA'))

      return availableSMResults(type, eventType, hasKVA)
    }
    return []
  }
  return unique(
    requirements.rules.flatMap((rule) =>
      asArray(rule)
        .filter((opt) => !type || opt.type === type)
        .map((opt) => (opt.cert ? 'CERT' : opt.result))
    )
  )
}

export function createMissingResult(
  requirements: EventResultRequirementsByDate | undefined,
  results: Array<QualifyingResult | ManualTestResult>,
  regNo: string
): ManualTestResult {
  const rule = findFirstMissing(requirements, results)
  return {
    id: nanoid(10),
    regNo,
    date: new Date(),
    official: false,
    qualifying: true,
    type: '',
    judge: '',
    location: '',
    result: '',
    class: '',
    ...rule,
  }
}

export function resultBorderColor(qualifying: boolean | undefined) {
  if (qualifying === true) {
    return 'success.light'
  }
  if (qualifying === false) {
    return 'error.main'
  }
}

export function getResultId(result: ManualTestResult | QualifyingResult) {
  if ('id' in result) {
    return result.id
  }
  return result.type + result.class + result.date.toISOString()
}
