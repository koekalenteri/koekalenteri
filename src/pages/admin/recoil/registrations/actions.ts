import type { PublicDogEvent, Registration, RegistrationGroupInfo } from '../../../../types'

import { useSnackbar } from 'notistack'
import { useRecoilState, useRecoilValue } from 'recoil'

import { createRefund } from '../../../../api/payment'
import {
  getRegistrationTransactions,
  putAdminRegistration,
  putAdminRegistrationNotes,
  putRegistrationGroups,
} from '../../../../api/registration'
import { reportError } from '../../../../lib/client/error'
import { idTokenAtom } from '../../../recoil'
import { adminEventSelector } from '../events'

import { currentEventRegistrationsSelector } from './selectors'

export const useAdminRegistrationActions = (eventId: string) => {
  const [adminRegistrations, setAdminRegistrations] = useRecoilState(currentEventRegistrationsSelector)
  const [event, setEvent] = useRecoilState(adminEventSelector(eventId))
  const token = useRecoilValue(idTokenAtom)
  const { enqueueSnackbar } = useSnackbar()

  const updateAdminRegistration = (saved: Registration) => {
    const regs = [...adminRegistrations]
    const index = regs.findIndex((r) => r.id === saved.id)
    const insert = index === -1
    regs.splice(insert ? regs.length : index, insert ? 0 : 1, saved)
    setAdminRegistrations([...regs])
  }

  return {
    async save(reg: Registration) {
      if (!token) throw new Error('missing token')
      const regWithOverrides = {
        ...reg,
        handler: reg.ownerHandles ? { ...reg.owner } : reg.handler,
        payer: reg.ownerPays ? { ...reg.owner } : reg.payer,
      }
      const saved = await putAdminRegistration(regWithOverrides, token)
      updateAdminRegistration(saved)
      return saved
    },

    async saveGroups(eventId: string, groups: RegistrationGroupInfo[]) {
      try {
        if (!token) throw new Error('missing token')
        const {
          items,
          classes,
          entries,
          invitedOk,
          invitedFailed,
          pickedOk,
          pickedFailed,
          reserveOk,
          reserveFailed,
          cancelledOk,
          cancelledFailed,
        } = await putRegistrationGroups(eventId, groups, token)

        if (pickedOk.length) {
          enqueueSnackbar('Koepaikkailmoitus lÃ¤hetetty onnistuneesti\n\n' + pickedOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (invitedOk.length) {
          enqueueSnackbar('Koekutsu lÃ¤hetetty onnistuneesti\n\n' + invitedOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (reserveOk.length) {
          enqueueSnackbar('Varasijailmoitus lÃ¤hetetty onnistuneesti\n\n' + reserveOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (cancelledOk.length) {
          enqueueSnackbar('Peruutusilmoitus lÃ¤hetetty onnistuneesti\n\n' + reserveOk.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (pickedFailed.length) {
          enqueueSnackbar('Koepaikkailmoituksen lÃ¤hetys epÃ¤onnistui ðŸ’©\n\n' + pickedFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (invitedFailed.length) {
          enqueueSnackbar('Koekutsun lÃ¤hetys epÃ¤onnistui ðŸ’©\n\n' + invitedFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (pickedFailed.length) {
          enqueueSnackbar('Varasijailmoituksen lÃ¤hetys epÃ¤onnistui ðŸ’©\n\n' + reserveFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        if (cancelledFailed.length) {
          enqueueSnackbar('Peruutusilmoitukse lÃ¤hetys epÃ¤onnistui ðŸ’©\n\n' + reserveFailed.join('\n'), {
            variant: 'success',
            style: { whiteSpace: 'pre-line' },
          })
        }
        setAdminRegistrations(items)
        if (event) {
          setEvent({ ...event, classes, entries })
        }
      } catch (e) {
        reportError(e)
        return false
      }
    },

    async transactions(eventId: PublicDogEvent['id'], registrationId: Registration['id']) {
      if (!token) throw new Error('missing token')

      return getRegistrationTransactions(eventId, registrationId, token)
    },

    async refund(reg: Registration, transactionId: string, amount: number) {
      if (!token) throw new Error('missing token')

      const result = await createRefund(transactionId, amount, token)

      if (result?.status === 'pending' || result?.status === 'ok') {
        updateAdminRegistration({
          ...reg,
          refundAt: new Date().toISOString(),
          refundStatus: result?.status === 'pending' || result.provider === 'email refund' ? 'PENDING' : 'SUCCESS',
          refundAmount: (reg.refundAmount ?? 0) + amount / 100,
        })
      }

      return result
    },

    async putInternalNotes(
      eventId: Registration['eventId'],
      id: Registration['id'],
      internalNotes: Registration['internalNotes']
    ) {
      if (!token) throw new Error('missing token')

      const reg = adminRegistrations.find((r) => r.id === id)
      if (!reg) throw new Error('unexpected error occured')

      await putAdminRegistrationNotes({ eventId, id, internalNotes }, token)
      updateAdminRegistration({ ...reg, internalNotes })
    },
  }
}
