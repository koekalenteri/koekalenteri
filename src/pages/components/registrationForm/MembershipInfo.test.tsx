import type { ReactNode } from 'react'
import type { Registration } from '../../../types'

import { Suspense } from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers'
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFnsV3'
import { render, screen } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { RecoilRoot } from 'recoil'

import { registrationWithStaticDates } from '../../../__mockData__/registrations'
import { locales } from '../../../i18n'
import { clone } from '../../../lib/utils'
import { flushPromises, renderWithUserEvents } from '../../../test-utils/utils'

import MembershipInfo from './MembershipInfo'

jest.mock('../../../api/dog')
jest.mock('../../../api/registration')

function Wrapper(props: { readonly children?: ReactNode }) {
  return (
    <LocalizationProvider dateAdapter={AdapterDateFns} adapterLocale={locales.fi}>
      <RecoilRoot>
        <SnackbarProvider>
          <Suspense fallback={<div>loading...</div>}>{props.children}</Suspense>
        </SnackbarProvider>
      </RecoilRoot>
    </LocalizationProvider>
  )
}
describe('MembershipInfo', () => {
  beforeAll(() => jest.useFakeTimers())
  afterEach(() => jest.runOnlyPendingTimers())
  afterAll(() => jest.useRealTimers())

  it('should render with minimal info', () => {
    const { container } = render(<MembershipInfo reg={{}} orgId={'test'} />, { wrapper: Wrapper })
    expect(container).toMatchSnapshot()
  })

  it('should render when owner handles and is member', () => {
    const reg = clone<Registration>(registrationWithStaticDates)
    reg.ownerHandles = true
    reg.owner!.membership = true

    const { container } = render(<MembershipInfo reg={reg} orgId={'test'} />, { wrapper: Wrapper })
    expect(container).toMatchSnapshot()
  })

  it('should call onChange', async () => {
    const reg = clone<Registration>(registrationWithStaticDates)
    reg.ownerHandles = false
    reg.owner!.membership = false
    reg.handler!.membership = false

    const onChange = jest.fn((props) => Object.assign(reg, props))
    const { user } = renderWithUserEvents(
      <MembershipInfo reg={reg} orgId={'test'} onChange={onChange} />,
      { wrapper: Wrapper },
      {
        advanceTimers: jest.advanceTimersByTime,
      }
    )

    await flushPromises()
    expect(onChange).not.toHaveBeenCalled()

    const ownerCheckbox = screen.getByRole('checkbox', { name: 'registration.ownerIsMember' })
    const hadnlerCheckbox = screen.getByRole('checkbox', { name: 'registration.handlerIsMember' })

    await user.click(ownerCheckbox)
    await flushPromises()
    expect(onChange).toHaveBeenLastCalledWith({
      owner: {
        ...reg.owner,
        membership: true,
      },
    })

    await user.click(hadnlerCheckbox)
    await flushPromises()
    expect(onChange).toHaveBeenLastCalledWith({
      handler: {
        ...reg.handler,
        membership: true,
      },
    })

    await flushPromises()
  })
})
