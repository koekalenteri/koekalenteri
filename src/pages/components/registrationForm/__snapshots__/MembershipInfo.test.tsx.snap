// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembershipInfo should render when owner handles and is member 1`] = `
<div>
  <div
    class="MuiBox-root css-l530ik"
  >
    <button
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-10rc6rd-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      id=":r1:"
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="KeyboardArrowDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
        />
      </svg>
    </button>
    <div
      class="MuiBox-root css-1qbrvqz"
    >
      <div
        class="MuiBox-root css-1jw9kr7"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          registration.membership
        </p>
        <p
          class="MuiFormHelperText-root css-1b0t562-MuiFormHelperText-root"
        />
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-qr6njo-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiBox-root css-e1h6et"
            >
              <div
                class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-j8yymo"
                      data-indeterminate="false"
                      name="ownerIsMember"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    registration.ownerIsMember
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root Mui-disabled MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="true"
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-disabled PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked Mui-disabled MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
                    tabindex="-1"
                  >
                    <input
                      checked=""
                      class="PrivateSwitchBase-input css-j8yymo"
                      data-indeterminate="false"
                      disabled=""
                      name="handlerIsMember"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label Mui-disabled css-rizt0-MuiTypography-root"
                  >
                    registration.handlerIsMember
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MembershipInfo should render with minimal info 1`] = `
<div>
  <div
    class="MuiBox-root css-l530ik"
  >
    <button
      class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-10rc6rd-MuiButtonBase-root-MuiIconButton-root"
      disabled=""
      id=":r0:"
      tabindex="-1"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="KeyboardArrowRightIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6z"
        />
      </svg>
    </button>
    <div
      class="MuiBox-root css-1qbrvqz"
    >
      <div
        class="MuiBox-root css-1jw9kr7"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
        >
          registration.membership
        </p>
        <p
          class="MuiFormHelperText-root Mui-error css-1snnebk-MuiFormHelperText-root"
        >
          validation.registration.choose field
        </p>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-cwrbtg-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-1x6hinx-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-1i4ywhz-MuiCollapse-wrapperInner"
          >
            <div
              class="MuiBox-root css-e1h6et"
            >
              <div
                class="MuiFormGroup-root css-1mac7pt-MuiFormGroup-root"
              >
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-j8yymo"
                      data-indeterminate="false"
                      name="ownerIsMember"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    registration.ownerIsMember
                  </span>
                </label>
                <label
                  class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
                >
                  <span
                    class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-imrjgg-MuiButtonBase-root-MuiCheckbox-root"
                  >
                    <input
                      class="PrivateSwitchBase-input css-j8yymo"
                      data-indeterminate="false"
                      name="handlerIsMember"
                      type="checkbox"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="CheckBoxOutlineBlankIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
                  >
                    registration.handlerIsMember
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
